### ISSUE TITLE
Display errors closer to the place where fix should go

### ISSUE THREAD

**Issue by nightwing** [eslint/eslint#4470]
`space-after-keywords` and  `no-sequences` rules display errors on wrong lines for the following snippet
```js
/*eslint "space-after-keywords": 2, "no-sequences": 2 */
do { // error for while is here
    
} while(true)

a( // error for , is here
{
}
),
b
```

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.
